cs/src/core/io/IntegerHelper.cs

```diff
@@ -65,6 +111,65 @@ public static int EncodeVarUInt32(byte[] data, uint value, int index)
             return index;
         }
 
+        public static int GetVarUInt64Length(ulong value)
+        {
+            int len = 1;
+            // byte 0
+            if (value >= 0x80)
+            {
+                len++;
+                value >>= 7;
+                // byte 1
+                if (value >= 0x80)
+                {
+                    len++;
+                    value >>= 7;
+                    // byte 2
+                    if (value >= 0x80)
+                    {
+                        len++;
+                        value >>= 7;
+                        // byte 3
+                        if (value >= 0x80)
+                        {
+                            len++;
+                            value >>= 7;
+                            // byte 4
+                            if (value >= 0x80)
+                            {
+                                len++;
+                                value >>= 7;
+                                // byte 5
+                                if (value >= 0x80)
+                                {
+                                    len++;
+                                    value >>= 7;
+                                    // byte 6
+                                    if (value >= 0x80)
+                                    {
+                                        len++;
+                                        value >>= 7;
+                                        // byte 7
+                                        if (value >= 0x80)
+                                        {
+                                            len++;
+                                            value >>= 7;
+                                            // byte 8
+                                            if (value >= 0x80)
+                                            {
+                                                len++;
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            return len;
+        }
```

Comment:
```c#
public static int GetVarUInt64Length(ulong value)  
{  
    int len = 1;
    while (value >= 0x80) {
        value >>= 7;
        len++;
    }
    return len;
}
```
In C#, when right-shifting uint or ulong, a logical-shift is performed, zero-filling the high bits.